server/update_branch.go

```diff
@@ -2,60 +2,68 @@ package server
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-mattermod/model"
 
 	"github.com/google/go-github/v32/github"
 )
 
-func (s *Server) handleUpdateBranch(ctx context.Context, eventIssueComment IssueComment) {
-	prGitHub, _, err := s.GithubClient.PullRequests.Get(ctx,
-		*eventIssueComment.Repository.Owner.Login,
-		*eventIssueComment.Repository.Name,
-		*eventIssueComment.Issue.Number,
-	)
-	if err != nil {
-		mlog.Error("Error getting the latest PR information from github", mlog.Err(err))
-		return
-	}
-	pr, err := s.GetPullRequestFromGithub(ctx, prGitHub)
-	if err != nil {
-		mlog.Error("Error Updating the PR in the DB", mlog.Err(err))
-		return
-	}
+var (
+	ErrCommenterPermission    = errors.New("commenter does not have permissions")
+	ErrOrganizationPermission = errors.New("we don't have permissions")
+	ErrUpdatePullRequest      = errors.New("could not update pull request")
+)
+
+const (
+	MsgCommenterPermission    = "Looks like you don't have permissions to trigger this command.\n Only available for the PR submitter and org members"
+	MsgOrganizationPermission = "We don't have permissions to update this PR, please contact the submitter to apply the update."
+	MsgUpdatePullRequest      = "Error trying to update the PR.\nPlease do it manually."
+)
```

Comment:
I have an idea to improve this further. Needs a little bit of more work, but might be more idiomatic. What we are doing here is essentially map the error to the string. So the contents of the error is entirely wasted here, as what gets printed is not the error string, but another message which is mapped to it.

Perhaps what we can do is have error types instead like

```go
type ErrPermission struct {
  source string // this can be commenter/org
}

func (e *ErrPermission) Error() string {
  // Now here we print the proper message depending on the source
}

func newErrPermission(source) *ErrPermission {
 // return the error
}
```

This way we can reduce the duplication in information in the errors and the string messages. Because basically we want to capture errors and then pass the error messages along.

Might be better to do this in a separate PR since a lot of work has already gone in to this one. If you think this is a good idea, let's file a ticket and work on it later.

Reply:
Agreed, it would be more convenient to handle errors/messages that way/ There are several occurrences like this in the `server` package that might benefit from this treatment. Created the ticket [here](https://mattermost.atlassian.net/browse/MM-26783) 👍